src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1144,97 +820,57 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
                 bool chunkedModeForSend = IsChunkedModeForSend(state.RequestMessage);
 
                 AddRequestHeaders(
-                    requestHandle,
+                    state.RequestHandle,
                     state.RequestMessage,
                     _cookieUsePolicy == CookieUsePolicy.UseSpecifiedCookieContainer ? _cookieContainer : null);
 
                 uint proxyAuthScheme = 0;
                 uint serverAuthScheme = 0;
-                bool retryRequest = false;
-
-                do
+                state.RetryRequest = false;
+
+                // The only way to abort pending async operations in WinHTTP is to close the WinHTTP handle.
```

Comment:
I expect that the comment applies for all async methods. Maybe you want to extract the cancellation handling in a separate method and keep the comment there. 

As a personal preference, this comment is too long to exist within the source-code. It will easily go out of sync with the code as the design changes.